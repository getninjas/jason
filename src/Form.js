import React from 'react';
import { Form, Field } from 'react-final-form';


const onSubmit = () => console.log('ok')
const validate = () => console.log('ok')
const handleSubmit = () => console.log('ok')

const MyForm = ({data}) => {
  return (
    <Form
      render={() => {
        console.log(data);
        return (
          <form onSubmit={onSubmit}>
            <h2>Simple Default Input</h2>
            <div>
              <label>First Name</label>
              <Field name="firstName" component="input" placeholder="First Name" />
            </div>
          </form>
        );
      }}
    />
  );
};

export default MyForm;
