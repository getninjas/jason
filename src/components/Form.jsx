import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Step from './Step';
import Breadcrumb from './Breadcrumb';

export default class Form extends Component {
  constructor() {
    super();

    this.state =  {
      activeStep: 0,
    };
  }

  componentDidMount() {
    this.setState({ activeStep: 0 });
  }

  render() {
    const { action, method, name, form } = this.props;

    return (
      <section className="wall--inverted col-normal-8 col-small-12">
        <form action={action} method={method} name={name} className="form container">
          {
            form.steps.map((step, index) => {
              return <Step visible={this.state.activeStep === index} key={`step-${index}`} step={step} />
            })
          }
        </form>

        <Breadcrumb active={this.state.activeStep} steps={form.steps} />
      </section>
    );
  }
}

Form.defaultProps = {
  form: {
    steps: [],
  }
}

Form.propTypes = {
  action: PropTypes.string,
  method: PropTypes.string,
  name: PropTypes.string.isRequired,
  form: PropTypes.object.isRequired,
}
