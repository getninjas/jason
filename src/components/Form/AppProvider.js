import React, { Component } from 'react';
import PropTypes from 'prop-types';

const AppContext = React.createContext();

class AppProvider extends Component {
  state = {
    greeting : 'hello',
    sayHi: (greeting) => {
      this.setState({greeting: greeting})
    }
  }
  render() {
    return <AppContext.Provider value={this.state}>
    { this.props.children }
    </AppContext.Provider>
  }
}

AppProvider.propTypes = {
  children: PropTypes.any,
}

export default AppProvider;
