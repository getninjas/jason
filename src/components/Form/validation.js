import errorMessages from '../../../src/helpers/errorMessages';

export const isEmpty = (value) => {
  const regex = /^\s*$/;

  return regex.test(value.toString().trim());
};

export const isMinLength = (text, length) => {
  let result = true;

  if (length) {
    result = text.replace(/\s/gm, '').length >= length;
  }

  return result;
};

export const isValidEmail = (value) => {
  const regex = /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/i;
  return regex.test(value.trim());
};

export const isValidZipcode = (value) => {
  const regex = /^[0-9]{5}-[0-9]{3}$/;
  return regex.test(value.trim());
};

export const isValidCellPhone = (value) => {
  const regex = /^\d{2}[6-9]{1}[0-9]{8}$/g;
  return regex.test(value.replace(/\D+/g, '').trim());
};

export const validateField = ({ required, type, value, minLength }) => {
  if (required && value === null) {
    return errorMessages.REQUIRED_FIELD;
  }

  if (type === 'phone' && (isEmpty(value) || !isValidCellPhone(value))) {
    return errorMessages.REQUIRED_VALID_CELLPHONE;
  }

  if (type === 'email' && !isValidEmail(value)) {
    return errorMessages.REQUIRED_VALID_EMAIL;
  }

  if (type === 'zipcode' && (isEmpty(value) || !isValidZipcode(value))) {
    return errorMessages.REQUIRED_VALID_ZIPCODE;
  }

  if (required && isEmpty(value)) {
    return errorMessages.REQUIRED_FIELD;
  }

  if (required && !isMinLength(value, minLength)) {
    return errorMessages.REQUIRED_MINLENGHT(minLength);
  }

  return '';
};

export const validateStep = (fields) => {
  let isValid = true;

  const updatedFields = fields.map((field) => {
    const modifiedField = Object.assign({}, field);

    modifiedField.errorMessage = validateField(modifiedField);

    if (modifiedField.errorMessage.length) {
      isValid = false;
    }

    return modifiedField;
  });

  return { updatedFields, isValid };
};
