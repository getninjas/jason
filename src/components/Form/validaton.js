import validator from 'validator';

export const validateValue = ({ required, type, value }) => {
  if (required && value === null) {
    return 'This field is required';
  }

  if (type === 'phone' && validator.isEmpty(value)) {
    return 'Invalid phone';
  }

  if (type === 'email' && !validator.isEmail(value)) {
    return 'Invalid email';
  }

  if (required && validator.isEmpty(value)) {
    return 'This field is required';
  }

  return '';
}

export const validateStep = (fields)=> {
  let isValid = true;

  const updatedFields = fields.map((field) => {
    const modifiedField = Object.assign({}, field);
    const errorMessage = validateValue(modifiedField);

    if (errorMessage) {
      modifiedField.errorMessage = errorMessage;
      isValid = false;
    }

    return modifiedField;
  });

  return { updatedFields, isValid };
}
