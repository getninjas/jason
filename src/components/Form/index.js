import axios from 'axios';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { AppContext } from '../../AppContext';
import Breadcrumb from '../Breadcrumb';
import Step from '../Step';
import { validateField, validateStep } from './validation';

const propTypes = {
  name: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired,
  action: PropTypes.string.isRequired,
  method: PropTypes.string,
  onReady: PropTypes.func,
  handleZipcodeExternalLinkClick: PropTypes.func,
  onZipcodeFetchSuccess: PropTypes.func,
  onZipcodeFetchError: PropTypes.func,
  onSubmitSuccess: PropTypes.func,
  onSubmitFieldError: PropTypes.func,
  onSubmitError: PropTypes.func,
  onSubmit: PropTypes.func,
  onStepChange: PropTypes.func,
};

const defaultProps = {
  method: 'POST',
  onReady() {},
  handleZipcodeExternalLinkClick(event) { return false; },
  onZipcodeFetchSuccess() {},
  onZipcodeFetchError() {},
  onSubmit() {},
  onSubmitSuccess() {},
  onSubmitFieldError() {},
  onSubmitError() {},
  onStepChange() {},
};

export default class Form extends Component {
  constructor(props) {
    super(props);

    this.state = {
      activeStepIndex: 0,
      handleZipcodeExternalLinkClick: (event) => this.handleZipcodeExternalLinkClick(event),
      onZipcodeFetchSuccess: data => this.onZipcodeFetchSuccess(data),
      onZipcodeFetchError: data => this.props.onZipcodeFetchError(data),
      stepsCount: 0,
      address: {
        typeStreet: '',
        street: '',
        neighborhood: '',
        city: '',
        uf: '',
      },
      action: this.props.action,
      steps: [],
    };

    this.formStyle = 'form container sh-form-content space-box-small';
    this.onSubmit = this.onSubmit.bind(this);
    this.onFieldChange = this.onFieldChange.bind(this);
    this.onFieldBlur = this.onFieldBlur.bind(this);
    this.errorMessages = this.props.data.errorMessages;
    this.handleZipcodeExternalLinkClick = this.handleZipcodeExternalLinkClick.bind(this);
  }

  get currentStep() {
    const { steps, activeStepIndex } = this.state;

    return steps[activeStepIndex];
  }

  componentDidMount() {
    this.setState({
      activeStepIndex: 0,
      stepsCount: this.props.data.steps.length - 1,
      steps: this.props.data.steps,
    }, this.props.onReady);
  }

  updateState(state) {
    Object.assign(state, { ...this.state, action: state.action || this.props.action });

    this.setState({ ...state });
  }

  handleZipcodeExternalLinkClick(event) {
    const handleZipcodeExists = this.props && this.props.handleZipcodeExternalLinkClick;
    if (!handleZipcodeExists) { return false; }
    const { handleZipcodeExternalLinkClick } = this.props;
    return handleZipcodeExternalLinkClick(event);
  }

  onZipcodeFetchSuccess(data) {
    this.props.onZipcodeFetchSuccess(data);
    const { typeStreet, street, neighborhood, city, uf } = data;

    this.setState({
      address: {
        typeStreet,
        street,
        neighborhood,
        city,
        uf,
      },
    });
  }

  formSubmit() {
    this.handleStepChange();

    if (this.isLastStep(this.state.activeStepIndex)) {
      this.handleSubmit();
    }
  }

  onSubmit(evt) {
    evt.preventDefault();

    this.formSubmit();
  }

  handleSubmit() {
    if (this.isStepsValid()) {
      this.submitRequest();
    }
  }

  isStepsValid() {
    const errorMessages = this.errorMessages;
    const validSteps = this.state.steps.filter((step) => {
      const { isValid } = validateStep(step.fields, errorMessages);

      return isValid;
    });

    return validSteps.length === this.props.data.steps.length;
  }

  async submitRequest() {
    try {
      this.props.onSubmit();

      const body = this.getFields();
      const response = await axios.post(this.state.action, body);

      this.props.onSubmitSuccess(response);
    } catch (error) {
      this.props.onSubmitError(error);
    }
  }

  getFields() {
    const fields = this.state.steps.map(step => step.fields);

    return { data: { ...fields, address: { ...this.state.address } } };
  }

  beforeStepChange() {
    const currentStep = this.state.steps[this.state.activeStepIndex];

    return currentStep.beforeChange ? currentStep.beforeChange(currentStep) : this.props.onStepChange();
  }

  async handleStepChange() {
    const { updatedFields, isValid } = validateStep(this.currentStep.fields, this.errorMessages);

    this.updateStep(updatedFields);

    if (isValid) {
      await this.beforeStepChange();

      this.nextStep(this.state);

      return;
    }

    const step = this.state.activeStepIndex < this.state.stepsCount ? 'current' : 'last';
    this.props.onSubmitFieldError(step);
  }

  nextStep({ activeStepIndex, stepsCount }) {
    if (activeStepIndex < stepsCount) {
      this.setState({ activeStepIndex: activeStepIndex + 1 });
    }
  }

  onFieldBlur({ value, id, required, type, minLength, regexPattern }) {
    const errorMessages = this.errorMessages;
    const fields = this.currentStep.fields.map((item) => {
      if (item.id === id) {
        const errorMessage = validateField({
          minLength,
          regexPattern,
          required,
          type,
          value,
        }, errorMessages);

        return { ...item, value, errorMessage };
      }

      return item;
    });

    this.updateStep(fields);
  }

  onFieldChange({ value, id }) {
    const fields = this.currentStep.fields.map((item) => {
      if (item.reference === id) {
        const defaultValue = { values: [{ databaseId: '', value: item.mask }] };

        return { ...item, values: (item.nested_values[value] || defaultValue).values, value: null };
      }

      if (item.id === id) {
        return { ...item, value };
      }

      return item;
    });

    this.updateStep(fields);
  }

  updateStep(fields) {
    const { steps, activeStepIndex } = this.state;

    const modifiedSteps = [...steps];
    const modifiedStep = { ...modifiedSteps[activeStepIndex], fields };

    modifiedSteps[activeStepIndex] = modifiedStep;

    this.setState({ steps: modifiedSteps });
  }

  isStepVisible(index) {
    return this.state.activeStepIndex === index;
  }

  isLastStep(index) {
    return index === this.state.stepsCount;
  }

  render() {
    return (
      <AppContext.Provider value={this.state}>
        <form noValidate onSubmit={this.onSubmit} name={this.props.name} action={this.state.action}
          className={this.formStyle}>
          {
            this.state.steps.map((step, index) => {
              const { buttonText, headerMarkup, fields } = step;

              return (
                <Step
                  buttonText={buttonText}
                  fields={fields}
                  formName={this.props.name}
                  onSubmit={this.onSubmit}
                  headerMarkup={headerMarkup}
                  isLast={this.isLastStep(index)}
                  key={`${this.props.name}-step-${index}`}
                  onFieldBlur={this.onFieldBlur}
                  onFieldChange={this.onFieldChange}
                  visible={this.isStepVisible(index)}
                  zipcodeUrlService={this.props.data.zipcodeUrlService}
                />
              );
            })
          }
        </form>

        <Breadcrumb active={this.state.activeStepIndex} steps={this.state.steps} />
      </AppContext.Provider>
    );
  }
}

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;
