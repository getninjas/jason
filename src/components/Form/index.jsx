import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { AppContext } from '../../AppContext';
import Step from '../Step';
import Breadcrumb from '../Breadcrumb';
import { validateField, validateStep } from './validation';

const propTypes = {
  name: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired,
  action: PropTypes.string.isRequired,
  method: PropTypes.string,
};

const defaultProps = {
  method: 'POST',
};

export default class Form extends Component {
  constructor() {
    super();

    this.state = {
      activeStepIndex: 0,
      onZipcodeFetchSuccess: zipcode => window.alert(`zipcode fectch success hook ${zipcode}`),
      onZipcodeFetchError: zipcode => window.alert(`zipcode fectch error hook ${zipcode}`),
      stepsCount: 0,
      steps: [],
    };

    this.sectionStyle = "wall--inverted col-normal-8 col-small-12";
    this.formStyle = "form container sh-form-content space-box-small";

    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleButtonClick = this.handleButtonClick.bind(this);
    this.onFieldChange = this.onFieldChange.bind(this);
  }

  get currentStep() {
    const { steps, activeStepIndex } = this.state;

    return steps[activeStepIndex];
  }

  componentDidMount() {
    this.setState({
      activeStepIndex: 0,
      stepsCount: this.props.data.steps.length - 1,
      steps: this.props.data.steps,
    });
  }

  handleSubmit(evt) {
    evt.preventDefault();

    this.handleStepChange();
  }

  handleButtonClick(evt) {
    evt.preventDefault();

    this.handleStepChange();
  }

  handleStepChange() {
    const { updatedFields, isValid } = validateStep(this.currentStep.fields);

    this.updateStep(updatedFields);

    if (isValid) {
      this.nextStep(this.state);
    }
  }

  nextStep({ activeStepIndex, stepsCount }) {
    if (activeStepIndex < stepsCount) {
      this.setState({ activeStepIndex: activeStepIndex + 1 });
    }
  }

  onFieldChange({ value, id, required, type, minLength }) {
    const fields = this.currentStep.fields.map((item) => {
      const itemID = `${this.props.name}-${item.id}`;

      if (itemID === id) {
        const errorMessage = validateField({ required, type, value, minLength });

        return Object.assign({}, item, { value, errorMessage });
      }

      return item;
    });

    this.updateStep(fields);
  }

  updateStep(fields) {
    const { steps, activeStepIndex } = this.state;

    let modifiedSteps = [...steps];
    let modifiedStep = Object.assign({}, modifiedSteps[activeStepIndex], { fields });

    modifiedSteps[activeStepIndex] = modifiedStep;

    this.setState({
      steps: modifiedSteps,
    });
  }

  isStepVisible(index) {
    return this.state.activeStepIndex === index;
  }

  isLastStep(index) {
    return index === this.state.stepsCount;
  }

  render() {
    const { action, method, name } = this.props;

    return (
      <AppContext.Provider value={this.state}>
        <section className={this.sectionStyle}>
          <form noValidate onSubmit={this.handleSubmit} action={action} method={method} name={name} className={this.formStyle}>
            {
              this.state.steps.map((step, index) => {
                const { buttonText, headerMarkup, fields } = step;

                return (
                  <Step
                    buttonText={buttonText}
                    fields={fields}
                    formName={name}
                    handleButtonClick={this.handleButtonClick}
                    headerMarkup={headerMarkup}
                    isLast={this.isLastStep(index)}
                    key={`${name}-step-${index}`}
                    onFieldChange={this.onFieldChange}
                    visible={this.isStepVisible(index)}
                    zipcodeUrlService={this.props.data.zipcodeUrlService}
                  />
                )
              })
            }
          </form>

          <Breadcrumb active={this.state.activeStepIndex} steps={this.state.steps} />
        </section>
      </AppContext.Provider>
    );
  }
}

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;
