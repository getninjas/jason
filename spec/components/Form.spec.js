import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../../src/components/Form';
import data from '../../src/form.json';

describe('Form', () => {
  it('renders custom props', () => {
    const component = renderer.create(
      <Form name={'form'} form={data.form} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders defaultProps', () => {
    const component = renderer.create(
      <Form name={'form'} form={data.form} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
