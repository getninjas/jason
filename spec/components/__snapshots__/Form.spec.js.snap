// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders custom props 1`] = `
<section
  className="wall--inverted col-normal-8 col-small-12"
>
  <form
    action="http://uol.com.br"
    className="form container"
    method="POST"
    name="form"
    onSubmit={[Function]}
  >
    <fieldset
      className="form__container inputs"
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h4 class='widget__title section-title--high-contrast'>Explique o que você precisa<small>Peça orçamento grátis, online!</small></h4>",
          }
        }
      />
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="1_id"
        >
          Qual é o tipo de serviço que você precisa?
        </label>
        <select
          className="form__input"
          defaultValue={7120}
          id="1_name_select"
          name="1_name_select"
          required="false"
        >
          <option
            value=""
          >
            Placeholder
          </option>
          <option
            value={7117}
          >
            Construção
          </option>
          <option
            value={7118}
          >
            Reformas
          </option>
          <option
            value={7119}
          >
            Instalações
          </option>
          <option
            value={7120}
          >
            Troca
          </option>
          <option
            value={7121}
          >
            Outros
          </option>
        </select>
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="2_id"
        >
          Especifique melhor a metragem do serviço
        </label>
        <input
          className="form__input"
          id="2_name_text"
          name="2_name_text"
          onChange={[Function]}
          placeholder="Bora pra action"
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="3_id"
        >
          Informações adicionais
        </label>
        <textarea
          className="form__input"
          id="3_name_textarea"
          name="3_name_textarea"
          onChange={[Function]}
          placeholder="Descreva o que você precisa"
          required={false}
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <button
        className="btn btn--regular btn--high-contrast btn--fluid space-box-medium"
        onClick={[Function]}
        type="button"
      >
        Fazer um orçamento grátis
      </button>
    </fieldset>
    <fieldset
      className="form__container inputs"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div class='form__alert-message space-element-large'><img src='https://www.getninjas.com.br/assets/request/phone_call.svg' class='form__icon' alt='Estamos quase lá'><h2 class='form__info'>Estamos quase lá<small>Não perca tempo ligando para vários profissionais. Preencha os dados abaixo e <strong>nós encontraremos os melhores pra você!</strong></small></h2></div>",
          }
        }
      />
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_1"
        >
          Em que bairro será o serviço?
        </label>
        <input
          className="form__input"
          id="user_field_1"
          name="user_field_1"
          onChange={[Function]}
          placeholder=""
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_2"
        >
          Nome
        </label>
        <input
          className="form__input"
          id="user_field_2"
          name="user_field_2"
          onChange={[Function]}
          placeholder=""
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_3"
        >
          E-mail
        </label>
        <input
          className="form__input"
          id="user_field_3"
          name="user_field_3"
          onChange={[Function]}
          placeholder="Ex: nome@email.com"
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_4"
        >
          DDD + Celular
        </label>
        <input
          className="form__input"
          id="user_field_4"
          name="user_field_4"
          onChange={[Function]}
          placeholder=""
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <button
        className="btn btn--regular btn--high-contrast btn--fluid space-box-medium"
        onClick={[Function]}
        type="submit"
      >
        Finalizar
      </button>
    </fieldset>
  </form>
  <ol
    className="form__steps space-box-medium"
  >
    <li
      className="form__steps-item form__steps-item--active"
    >
      Seu pedido
    </li>
    <li
      className="form__steps-item "
    >
      Seus dados
    </li>
  </ol>
</section>
`;

exports[`Form renders defaultProps 1`] = `
<section
  className="wall--inverted col-normal-8 col-small-12"
>
  <form
    action="http://uol.com.br"
    className="form container"
    method="POST"
    name="form"
    onSubmit={[Function]}
  >
    <fieldset
      className="form__container inputs"
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h4 class='widget__title section-title--high-contrast'>Explique o que você precisa<small>Peça orçamento grátis, online!</small></h4>",
          }
        }
      />
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="1_id"
        >
          Qual é o tipo de serviço que você precisa?
        </label>
        <select
          className="form__input"
          defaultValue={7120}
          id="1_name_select"
          name="1_name_select"
          required="false"
        >
          <option
            value=""
          >
            Placeholder
          </option>
          <option
            value={7117}
          >
            Construção
          </option>
          <option
            value={7118}
          >
            Reformas
          </option>
          <option
            value={7119}
          >
            Instalações
          </option>
          <option
            value={7120}
          >
            Troca
          </option>
          <option
            value={7121}
          >
            Outros
          </option>
        </select>
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="2_id"
        >
          Especifique melhor a metragem do serviço
        </label>
        <input
          className="form__input"
          id="2_name_text"
          name="2_name_text"
          onChange={[Function]}
          placeholder="Bora pra action"
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="3_id"
        >
          Informações adicionais
        </label>
        <textarea
          className="form__input"
          id="3_name_textarea"
          name="3_name_textarea"
          onChange={[Function]}
          placeholder="Descreva o que você precisa"
          required={false}
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <button
        className="btn btn--regular btn--high-contrast btn--fluid space-box-medium"
        onClick={[Function]}
        type="button"
      >
        Fazer um orçamento grátis
      </button>
    </fieldset>
    <fieldset
      className="form__container inputs"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div class='form__alert-message space-element-large'><img src='https://www.getninjas.com.br/assets/request/phone_call.svg' class='form__icon' alt='Estamos quase lá'><h2 class='form__info'>Estamos quase lá<small>Não perca tempo ligando para vários profissionais. Preencha os dados abaixo e <strong>nós encontraremos os melhores pra você!</strong></small></h2></div>",
          }
        }
      />
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_1"
        >
          Em que bairro será o serviço?
        </label>
        <input
          className="form__input"
          id="user_field_1"
          name="user_field_1"
          onChange={[Function]}
          placeholder=""
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_2"
        >
          Nome
        </label>
        <input
          className="form__input"
          id="user_field_2"
          name="user_field_2"
          onChange={[Function]}
          placeholder=""
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_3"
        >
          E-mail
        </label>
        <input
          className="form__input"
          id="user_field_3"
          name="user_field_3"
          onChange={[Function]}
          placeholder="Ex: nome@email.com"
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <div
        className="form__field form__field--fluid input"
      >
        <label
          className="form__label"
          htmlFor="user_field_4"
        >
          DDD + Celular
        </label>
        <input
          className="form__input"
          id="user_field_4"
          name="user_field_4"
          onChange={[Function]}
          placeholder=""
          required="false"
          title=""
          value=""
        />
        <span
          className="form__message form__message--invalid space-element-small error"
        >
          dsds
        </span>
      </div>
      <button
        className="btn btn--regular btn--high-contrast btn--fluid space-box-medium"
        onClick={[Function]}
        type="submit"
      >
        Finalizar
      </button>
    </fieldset>
  </form>
  <ol
    className="form__steps space-box-medium"
  >
    <li
      className="form__steps-item form__steps-item--active"
    >
      Seu pedido
    </li>
    <li
      className="form__steps-item "
    >
      Seus dados
    </li>
  </ol>
</section>
`;
